docs/api.md

```diff
@@ -248,6 +248,196 @@ after calling this method.
+```
+
+We don't guarantee backwards compatibility between beta versions.
+:::
+
+The authentication API is exported under the `auth` namespace. It allows you to authenticate users visiting your website via StackBlitz. In order for users to be authenticated via this method, they must:
+
+ * Be logged in on StackBlitz.
+ * Belong to the organisation you used to generate your `client_id` for use with the WebContainer API.
+ * Authorize your website.
+
+Once logged in, you'll be able to install private packages that those users have access to within WebContainer.
+
+### `auth` Functions
+
+<br />
+
+### ▸ `init`
+
+Intialize the authentication for use in WebContainer. This method should be call as soon as possible as part of the loading of your page. For example at the top of a module that gets loaded as soon as the page load. This is important for multiple reasons:
+
+ * If you do client side routing, and the OAuth flow is happening, then query parameters might be populated with values related to the OAuth flow. The `init` function remove them after they've been consumed.
+
+ * If you do the authentication in <a href="#▸-startauthflow">popup</a> mode, you probably want the popup to be closed as soon as the authentication completed to provide a great experience for your users.
+
+<h4 id="auth-init-signature">
+  <a id="auth-init-signature">Signature</a>
+  <a href="#auth-init-signature" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+<code>init(options: <a href="#authinit-options">AuthInitOptions</a>): { status: 'need-auth' | 'authorized' } | <a href="#authfailed-error">AuthFailedError</a></code>
+
+
+<h4 id="authinit-options">
+  <a id="authinit-options"><code>AuthInitOptions</code></a>
+  <a href="#authinit-options" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+```ts
+interface AuthInitOptions {
+  /**
+   * StackBlitz' origin.
+   *
+   * @default https://stackblitz.com
+   */
+  editorOrigin?: string;
+
+  /**
+   * The client id for this OAuth application.
+   */
+  clientId: string;
+
+  /**
+   * OAuth scope.
+   *
+   * @see https://www.rfc-editor.org/rfc/rfc6749#section-3.3
+   */
+  scope: string;
+}
+```
+
+<h4 id="authfailed-error">
+  <a id="authfailed-error"><code>AuthFailedError</code></a>
+  <a href="#authfailed-error" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+```ts
+interface AuthFailedError {
+  status: 'auth-failed';
+
+  /**
+   * A short description of the error.
+   */
+  error: string;
+
+  /**
+   * A detailed description of the error.
+   */
+  description: string;
+}
+```
+
+### ▸ `startAuthFlow`
+
+This starts the OAuth flow, redirecting the current page to the StackBlitz editor to authenticate the user unless `popup` is set to true in which case it's done in a popup.
+
+<code>startAuthFlow(options?: { popup?: boolean }): void</code>
+
+### ▸ `loggedIn`
+
+Returns a promise that resolves when the user authorized your application.
+This promise is guaranteed to never be rejected.
```

Comment:
Just out of curiosity. Why do we sometimes have long sentences on a single line? And why do we sometimes cut them off?

Reply:
Oops I didn't mean to do that, what do you think should be the preferred way?